src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1187,6 +1187,46 @@ private static DateTime ConvertToUTC(DateTime dateTime, TimeZoneInfo fromTimeZon
             return resultDateTime.UtcDateTime;
         }
 
+        public static FormulaValue UniChar(IRContext irContext, NumberValue[] args)
+        {
+            // Excel floors the input number
+            double number = Math.Floor(args[0].Value);
+
+            // Unicode upper and lower bounds
+            const double lowerBound = 0x1; // Excel returns error for UniChar(0)
+            const double upperBound = 0x10FFFF; // 1114111

```

Comment:
Any reason why we are not checking these bounds at compilation time?

Reply:
The argument may not be a number; it may be an expression or a variable, so we don't know it in all cases. We could check it at compile time if we have a literal number argument, but we still need a runtime check.

Reply:
Make sense. There is no need for a compilation check in this case.